src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/ValidateCredentials.java

```diff
@@ -1,28 +1,41 @@
 package io.stargate.sgv2.jsonapi.service.resolver.model.impl;
 
-import io.quarkus.grpc.GrpcClient;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.quarkus.runtime.ShutdownEvent;
 import io.stargate.embedding.gateway.EmbeddingGateway;
 import io.stargate.embedding.gateway.EmbeddingServiceGrpc;
 import io.stargate.sgv2.jsonapi.api.request.DataApiRequestInfo;
 import io.stargate.sgv2.jsonapi.exception.ErrorCode;
 import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
 import jakarta.inject.Inject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
 
 @ApplicationScoped
 public class ValidateCredentials {
-  @GrpcClient("embedding")
-  EmbeddingServiceGrpc.EmbeddingServiceBlockingStub embeddingService;
+
+  private @ConfigProperty(name = "quarkus.grpc.clients.embedding.host") String host;
+
+  @ConfigProperty(name = "quarkus.grpc.clients.embedding.port")
+  private int port;
+
+  private ManagedChannel channel = null;
 
   @Inject DataApiRequestInfo dataApiRequestInfo;
 
   public boolean validate(String provider, String value) {
+    if (channel == null) {
```

Comment:
how about synchronization? Could make `channel` synchronized; if this is used concurrently is it not necessary to consider synchronization?